core/port/range.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package port
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Range represents a networking port range
+type Range struct {
+	Start, End int
+}
+
+// UnspecifiedRange creates an unspecified range
+func UnspecifiedRange() *Range {
+	return &Range{0, 0}
+}
+
+// ParseRange parses port range expression, e.g. "1000:1200" into Range struct
+func ParseRange(rangeExpr string) (*Range, error) {
+	if rangeExpr == "" {
+		return UnspecifiedRange(), nil
+	}
+	bounds := strings.FieldsFunc(rangeExpr, func(c rune) bool {
+		return c == ':'
+	})
+	if len(bounds) != 2 {
+		return nil, errors.New("Invalid port range expression: " + rangeExpr)
```

Comment:
[Error strings should not be capitalized](https://github.com/golang/go/wiki/CodeReviewComments#error-strings):
```suggestion
		return nil, errors.New("invalid port range expression: " + rangeExpr)
```